moab_action_transform.ink

```diff
@@ -0,0 +1,132 @@
+###
+
+# MSFT Bonsai 
+# Copyright 2020 Microsoft
+# This code is licensed under MIT license (see LICENSE for details)
+
+# Moab Sample Using Action Transform during Training Only
+
+###
+
+inkling "2.0"
+
+using Math
+using Goal
+
+# Distances measured in meters
+const RadiusOfPlate = 0.1125 # m
+
+# Velocities measured in meters per sec.
+const MaxVelocity = 6.0
+const MaxInitialVelocity = 1.0
+
+# Threshold for ball placement
+const CloseEnough = 0.02
+
+# Default time delta between simulation steps (s)
+const DefaultTimeDelta = 0.045
+
+# Maximum distance per step in meters
+const MaxDistancePerStep = DefaultTimeDelta * MaxVelocity
+
+# Maximum plate angle in degrees
+const MaxPlateAngle = 22
+
+# State received from the simulator after each iteration
+type ObservableState {
+    # Ball X,Y position
+    ball_x: number<-MaxDistancePerStep - RadiusOfPlate .. RadiusOfPlate + MaxDistancePerStep>,
+    ball_y: number<-MaxDistancePerStep - RadiusOfPlate .. RadiusOfPlate + MaxDistancePerStep>,
+
+    # Ball X,Y velocity
+    ball_vel_x: number<-MaxVelocity .. MaxVelocity>,
+    ball_vel_y: number<-MaxVelocity .. MaxVelocity>,
+}
+
+type BrainAction {
```

Comment:
Suggest a comment here.

Reply:
Action provided as output by policy and sent as
# input to action transform